pyvelox/pyvelox.h

```diff
@@ -504,6 +532,75 @@ static void addVectorBindings(
             std::move(baseVector),
             PyVeloxContext::getSingletonInstance().pool());
       });
+
+  m.def(
+      "row_vector",
+      [](std::vector<std::string>& names,
+         std::vector<VectorPtr>& children,
+         const std::optional<py::dict>& nullabilityDict) {
+        if (children.size() == 0 || names.size() == 0) {
+          throw py::value_error("RowVector must have children.");
+        }
+        std::vector<std::shared_ptr<const Type>> childTypes;
+        childTypes.reserve(children.size());
+        size_t vectorSize = children.size() > 0 ? children[0]->size() : 0;
```

Comment:
In our case I think we should just assert children is not empty and then set vectorSize = children[0]->size().